### DISCUSSION TITLE
Pass data to pimcore_snippet

### DISCUSSION THREAD

**Discussion by RaederDev** [pimcore/pimcore#13630] [Q&A / Support]
Hi!

Is it possible to pass arbitrary data to a pimcore snippet?

I have a snippet that is positioned in a way that completely breaks the layout when embedded in a page that is currently in editmode. I would like to somehow make that snippet aware of the edit mode state of the page it's embedded in. Is that somehow possible? And if not would you be open to accept a MR that adds that feature to pimcore?

I'm thinking something like:
```twig
{{ pimcore_snippet("snippet", {defaultHeight: 400, myCustomField: editmode}) }}
```

Thanks!

**Comment by alexej-d** [comment]
You can grab the parameters in the snippet action inside its controller as follows:

```php
public function snippetAction(Request $request)
  {
    $defaultHeight = $request->get('defaultHeight');
    $myCustomField = $request->get('myCustomField');
    ...
  }
```
